modules/social_features/social_embed/src/Service/EmbedHelper.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\social_embed\Service;
+
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\file\Entity\File;
+
+/**
+ * Helper functions for Social Embed module.
+ */
+class EmbedHelper {
+
+  /**
+   * Module handler.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * Entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected EntityTypeManager $entityTypeManager;
+
+  /**
+   * Constructor for EmbedHelper.
+   *
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   Module handler.
+   * @param \Drupal\Core\Entity\EntityTypeManager $entity_type_manager
+   *   The entity manager object.
+   */
+  public function __construct(ModuleHandlerInterface $module_handler, EntityTypeManager $entity_type_manager) {
+    $this->moduleHandler = $module_handler;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * Returns file path of placeholder image.
+   *
+   * @return string
+   *   File path.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function getEmbedPlaceholderImage(): string {
+    /** @var \Drupal\file\FileInterface[] $files */
+    $files = $this->entityTypeManager->getStorage('file')->loadByProperties(['uri' => 'public://video_placeholder_img1.jpeg']);
```

Comment:
Didnt' we want to use something like markup instead of a image? So we can help the user with what they are actually giving consent to? (Our future front-end can decide for themselves what placeholder they want to render, they will know what URL needs to be part of the query in order to replace it)